[Feat] 메일함, 편지 기능 및 디자인 구현 마무리

markFT39 opened the pull request:
## 📝작업 내용

> 공감한 일기의 시간 표시 및 정렬 기준 변경
> 일기 공감 시 DB와 Local에 저장하는 로직 추가
> 메일함의 컨텐츠 열람 페이지 디자인
> 메일함 구현 완료

### 스크린샷 (선택)

<img src="https://github.com/user-attachments/assets/de544e58-a491-4c8a-bb44-914ffa12d3f5" alt="Resized Image" width="300">
<img src="https://github.com/user-attachments/assets/0ccf1207-026e-4c0f-a0d0-340db0196b18" alt="Resized Image" width="300">

## 💬리뷰 요구사항(선택)

> 열람 페이지의 화면 비율, 짤림 등이 기기 별로 문제가 없는지 리뷰 부탁 드립니다.

markFT39 committed:
[Feat] 메일함 시간 표시 및 정렬 수정

공감한 일기 추가한 날짜 기준 표시 및 정렬
everyMail 역시 동일한 규칙 적용

markFT39 committed:
[Feat] 일기 공감시 DB, Local 저장

공감 일기 반응시에 데이터 DB와 Local에 추가

markFT39 committed:
[Feat] 메일함 열람 기능 추가

편지, 공감한 일기 열람 기능 추가

KimGyeongLock reviewed (commented):
1. 위에다 stack으로 그대로 띄우니까 글자가 겹쳐져서 디자인이 별로네 -> 이건 디자이너랑 얘기해서 뒷배경 없애는 식으로 해도될듯
2. 그리고 보관함 text 위에 원래 공간이 별로 없는지 궁금함, 좀 좁아보이는 것 같기도 해서
3. 기쁘다 앞에 빈칸이 들어간건가 왜 저렇게 뜨지?
<img width="49" alt="image" src="https://github.com/user-attachments/assets/bb0c157c-511b-4478-95c7-d387d0b04bb4">

markFT39 commented:
1. 디자인에서 메일함 열람하는 디자인이 없어서 제가 임의로 편지 열람 디자인을 가져다 썼습니다.
2. '나의 일기' 페이지와 동일한 여백을 기준으로 가져왔습니다. 좁아 보인다면 회의 때 말해봐야 할 듯하네요.
3. 제가 개발할 때 확인한 바로는 DB에 도미노 아이콘(?)이 들어가 있더라구요, 지금은 지워진 건지 안보이는데 보통 맥에서 한글 입력시에 오류로 가끔 나타나던데 그건 것 같습니다. 로컬 저장소 지우기 하고 다시 불러오면 없어질 겁니다.

markFT39 committed:
[Fix] 메일함 필터 버그 수정

공감한 일기 탭의 필터가 전체 탭에도 적용되는 문제 수정

markFT39 committed:
[Feat] 메일함 열람시 배경 제거, 뒤로가기 제어

메일함 열람시 뒷배경 투명화
메일함, ai chat 페이지에서 뒤로가기 제어로 어플 종료 방지

markFT39 commented:
## 📝 추가 작업 내용

<img src="https://github.com/user-attachments/assets/d31e9a65-a692-490f-b958-13e0c917d60e" alt="Resized Image" width="300">

> 메일함 열람시 뒷 배경 투명화 추가
> 뒤로가기 버튼 동작 제어(메일함, ai chat)로 의도치 않은 어플 종료 방지

markFT39 committed:
[Feat] Firebase Finction 환경 추가

markFT39 committed:
[Feat] Firebase Function Hello World!

Firebase Function 첫 함수 배포와 테스트 호출 화면

markFT39 committed:
Firebase Function 환경 추가

markFT39 committed:
Merge branch 'feature/letter' of https://github.com/Nein-to-Sick/bandi_official into feature/letter

markFT39 commented:
> Firebase Function 환경 설정과 이를 통해 배포된 Web 상에서 get으로 데이터를 가져오는 테스트 페이지를 구현해 봤습니다.
> 테스트로 배포된 Web 주소: https://helloworld-25xhwjbd5q-uc.a.run.app/ 
> (접속이 원활히 되는지 테스트 부탁 드립니다!)

<img src="https://github.com/user-attachments/assets/fd69d415-b9aa-42bc-910e-6eeda6eccecc" alt="Resized Image" width="300">
<img src="https://github.com/user-attachments/assets/7004f87d-ef5f-4f9e-83b8-0578c98e683c" alt="Resized Image" width="300">
<img src="https://github.com/user-attachments/assets/31f8380f-d9c1-4dfc-8c38-3cf156085087" alt="Resized Image" width="300">

> 메일함의 appBar의 action 버튼으로 진입합니다.

markFT39 committed:
[Chore] 메일함 Tab의 텍스트 크기 수정

FittedBox로 텍스트 짤림 방지

markFT39 committed:
[Update] gpt 모델 개선 및 수정

gpt3.5-turbo에서 gpt4o-mini와 그 fine-tunning 모델로 수정 진행
비용과 성능에서 개선이 이뤄질 것으로 예상

markFT39 committed:
Merge branch 'develop' into feature/letter

markFT39 committed:
[Feat] Firebase Function 스케줄러 배포 성공

google Cloud Scheduler에 등록 성공한 코드입니다. 실제 실행과 테스트는 추후에 진행할 예정입니다.

markFT39 committed:
Merge branch 'feature/letter' of https://github.com/Nein-to-Sick/bandi_official into feature/letter

markFT39 committed:
[Feat] 편지 작성 및 DB 저장 기능 추가

Firebase Function 기능으로 chatGPT 호출 후 편지 작성해 유저별로 메세지 업데이트 가능

추후 AI 모델 or 프롬프트 개선 필요

markFT39 committed:
[Chore] firebase messaging, 변수 수정

letter 작성 이후 로컬에서 변화를 감지하는 방법을 고민할 필요가 있음

 firebase messaging은 letter 작성 이후 알림을 보내기 위함. 추후에 적용 가능성 있음.
